src/main/java/io/anserini/collection/BibtexCollection.java

```diff
@@ -0,0 +1,137 @@
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+/**
+ * A BibTex document collection.
+ */
+public class BibtexCollection extends DocumentCollection<BibtexCollection.Document> {
+  private static final Logger LOG = LogManager.getLogger(BibtexCollection.class);
+
+  public BibtexCollection(Path path){
+    this.path = path;
+    this.allowedFileSuffix = new HashSet<>(Arrays.asList(".bib"));
+  }
+
+  @Override
+  public FileSegment<BibtexCollection.Document> createFileSegment(Path p) throws IOException {
+    return new Segment(p);
+  }
+
+  /**
+   * A file in a Bibtex collection, typically containing multiple documents.
+   */
+  public static class Segment extends FileSegment<BibtexCollection.Document> {
+    private Iterator<Map.Entry<Key, BibTeXEntry>> iterator = null; // iterator for JSON document array
+    private BibTeXDatabase database;
+
+    protected Segment(Path path) throws IOException {
+      super(path);
+      bufferedReader = new BufferedReader(new FileReader(path.toString()));
+      BibTeXParser bibtexParser;
+      try {
+        bibtexParser = new BibTeXParser();
+      } catch (TokenMgrException | ParseException e) {
+        LOG.error("Error: Could not initialize BibTeX parser" + e.getMessage());
+        throw new IOException(e);
+      }
+      try {
+       database = bibtexParser.parse(bufferedReader);
+      } catch (ParseException | TokenMgrException | ObjectResolutionException e) {
+        LOG.error("Error: Could not parse BibTeX" + e.getMessage()); 
+        throw new IOException(e);
+      }
+      Map<Key, BibTeXEntry> entryMap = database.getEntries();
+      iterator = entryMap.entrySet().iterator();
+    }
+
+    @Override
+    public void readNext() throws NoSuchElementException {
+      if (iterator.hasNext()) {
+        Map.Entry<Key, BibTeXEntry> entry = iterator.next();
+        bufferedRecord = new BibtexCollection.Document(entry);
+      } else {
+        throw new NoSuchElementException("Reached end of BibtexDatabase Entries iterator");
+      }
+    }
+  }
+
+  /**
+   * A document in a Bibtex collection.
+   */
+  public static class Document implements SourceDocument {
+    private String id;
+    private String contents;
+    private String type;
+    private BibTeXEntry bibtexEntry;
+
+    public Document(Map.Entry<Key, BibTeXEntry> entry) {
+      id = entry.getKey().toString();
+      Map<Key, Value> bibtexFields = entry.getValue().getFields();
+      String doctitle = bibtexFields.getOrDefault(new Key("title"), new StringValue("", StringValue.Style.QUOTED)).toUserString();
+      String docAbstract = bibtexFields.getOrDefault(new Key("abstract"), new StringValue("", StringValue.Style.QUOTED)).toUserString();
+      contents = doctitle + ". " + docAbstract;
```

Comment:
Don't need the period since it'll just get indexed away

Reply:
I added it because when you query. Isn't it useful to be able to read the `contents` fields. If it was just space, how do you tell the title and the abstract from the contents field? Are you saying that it's a field not meant for human consumption but for indexing aid?

Reply:
resolving per discussion: contents does not need to be readable. remove period in next PR